app/javascript/components/widget/widget-templates/points-percentage/component.js

```diff
@@ -0,0 +1,29 @@
+import React, { Fragment } from 'react';
+import PropTypes from 'prop-types';
+
+import Chart from 'components/recharts';
+
+import config from './config';
+
+const AccessPointsPercentage = ({ widgetData }) => {
+
+
+  if (!widgetData) return null;
+
+  const data = config.parse(widgetData);
```

Comment:
there is one thing I don't like about this approach. You are parsing the data on every render... It's up to you to change it

Reply:
siiiiigh